### PR TITLE
Update README.md to add intentions and revision policy

### LINKED ISSUE #16: Formalise releases and revision policy
We should decide when we should release and how to decide version/revision numbers.
Also, are PR like https://github.com/gbdev/hardware.inc/pull/15 entitled to a minor version bump (i.e. 4.4.1)?

### LINKED ISSUE #21: README needs to be updated
The README needs to report our intentions with the file and how we are now releasing new versions.

> This repo is an attempt at version control. This is simply a kind of git mirror of the various flatfiles out there. I haven't written any of the actual code myself.

### PR DESCRIPTION
**Pull request by coltongit**
Adds a short blurb about the revision policy gbdev uses for revisions of this file and the intentions of what gbdev aims to do with it.
Fixes #16
Fixes #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,15 @@ The file was originally created by Jeff Frohwein in 1997, who still hosts [his l
...
+Breaking changes, like those in [72ec03f](https://github.com/gbdev/hardware.inc/commit/72ec03f835e72be83a1ef189a4a9eac0fbdf39e2), receive a major version bump, while smaller changes, not meant to break anything, yet are important enough to require a feature addition, recieve a minor version bump.
 
 ## Contributors
 
 * Jones (created the original `hardware.inc`, now lost)
 * Carsten Sørensen (whose ideas Jeff based his file on)
 * Jeff Frohwein
 * AntonioND
-* BlitterObjectBob, tobiasvl, ISSOtm,  avivace, GreenAndEievui, QuinnPainter, rondnelson99, daid, Hacktix, sukus21, alvaro-cuesta, basxto
+* BlitterObjectBob, tobiasvl, ISSOtm, avivace, GreenAndEievui, QuinnPainter, rondnelson99, daid, Hacktix, sukus21, alvaro-cuesta, basxto
```
Comment (eievui5) [inline README.md:19]:
```suggestion
* BlitterObjectBob, tobiasvl, ISSOtm, avivace, Eievui, QuinnPainter, rondnelson99, daid, Hacktix, sukus21, alvaro-cuesta, basxto
```
May as well change this while I'm reviewing :)

File: README.md [hunk 2]
```diff
@@ -1,18 +1,20 @@
 # `hardware.inc`
 ### Gameboy Hardware definitions
-`hardware.inc` has been the standard include file containing Game Boy hardware definitions for use in [rgbds](https://github.com/gbdev/rgbds) projects for over 20 years.
+`hardware.inc` has been the standard include file containing Game Boy hardware definitions for use in [rgbds](https://github.com/gbdev/rgbds) projects for over 20 years. Our intent here at gbdev is to continue to use `hardware.inc` as the standard as it has been for all these years.
```
Comment (ISSOtm) [inline README.md:3]:
```suggestion
`hardware.inc` has been the standard include file containing Game Boy hardware definitions for use in [RGBDS](https://rgbds.gbdev.io) projects for over 20 years.
```
I think that this is self-obvious enough, and if it really needs to be said, it should be said after the historical lines below?

File: README.md [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 
 The file was originally created by Jeff Frohwein in 1997, who still hosts [his latest version (2.3)](http://www.devrs.com/gb/files/hardware.zip) on his great [Dev'rs](http://devrs.com) website.
 
 Although Jeff tried to version control the file with a rudimentary change log at the top of the file, people have added small changes throughout the years, often without bumping the version number.
 
-This repo is an attempt at version control, using [@AntonioND](http://github.com/AntonioND) fork as baseline.
+This repo is an attempt at version control, using [@AntonioND](http://github.com/AntonioND)'s fork as the baseline.
```
Comment (ISSOtm) [inline README.md:9]: Can we call it the new reference, at this point?

File: README.md [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+
```
Comment (ISSOtm) [inline README.md:10]:
```suggestion

## Contributing

```

File: README.md [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
+Breaking changes, like those in [72ec03f](https://github.com/gbdev/hardware.inc/commit/72ec03f835e72be83a1ef189a4a9eac0fbdf39e2) receive a major version bump, while smaller changes which add new features but do not make any breaking changes receive a minor version bump.
```
Comment (ISSOtm) [inline README.md:11]: This is tentative, so feel free to disagree with and debate the following:
```suggestion
This repository's `master` branch should be considered production.
That is, each commit represents a new release, which thus needs the embedded version number bumped (and an entry added to the changelog at the beginning of the file).

We follow [semantic versioning](https://semver.org).
That is, breaking changes (such as those in [72ec03f](https://github.com/gbdev/hardware.inc/commit/72ec03f835e72be83a1ef189a4a9eac0fbdf39e2)) increase the major version, backwards-compatible changes (typically additions) only increase the minor version, and bugfixes only increase the patch level.

Changes to the comments should be considered a bugfix.
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ISSOtm) [reply README.md]: Should we add links to everyone who can? Should we instead just link to [the contributors page](https://github.com/gbdev/hardware.inc/graphs/contributors) for those who can?

Comment (eievui5) [reply README.md]: We could always do both :)

File: README.md [hunk 6]
```diff
@@ -1,18 +1,28 @@
...
 ### Gameboy Hardware definitions
-`hardware.inc` has been the standard include file containing Game Boy hardware definitions for use in [rgbds](https://github.com/gbdev/rgbds) projects for over 20 years.
+`hardware.inc` has been the standard include file containing Game Boy hardware definitions for use in [RGBDS](https://rgbds.gbdev.io) projects for over 20 years.
 
 The file was originally created by Jeff Frohwein in 1997, who still hosts [his latest version (2.3)](http://www.devrs.com/gb/files/hardware.zip) on his great [Dev'rs](http://devrs.com) website.
 
 Although Jeff tried to version control the file with a rudimentary change log at the top of the file, people have added small changes throughout the years, often without bumping the version number.
 
-This repo is an attempt at version control, using [@AntonioND](http://github.com/AntonioND) fork as baseline.
+This repo is an attempt at version control, using [@AntonioND](http://github.com/AntonioND)'s fork as the baseline. It has become the new official reference for documentation of the Game Boy and the underlying hardware.
```
Comment (ISSOtm) [inline README.md:9]:
```suggestion
This repo is an attempt at version control, using [@AntonioND](http://github.com/AntonioND)'s fork as the baseline.
It has become the new official reference for documentation of the Game Boy and the underlying hardware.
```
I'll also update the version-checking macro accordingly.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (ISSOtm) [reply README.md:9]: If this repo is the new reference for `hardware.inc` (not to the GB hardware! That's Pan Docs), then it shouldn't be called "an attempt at version control" anymore. We should however keep that it is based on AntonioND's fork.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (coltongit) [reply README.md]: Edited in latest commit to reflect this

**Comment by ISSOtm** [comment]
@avivace I updated `rev_Check_hardware_inc` to perform a SemVer compatibiliity check, please tell me if I got it wrong

**Review by ISSOtm — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,18 +1,28 @@
...
 The file was originally created by Jeff Frohwein in 1997, who still hosts [his latest version (2.3)](http://www.devrs.com/gb/files/hardware.zip) on his great [Dev'rs](http://devrs.com) website.
 
 Although Jeff tried to version control the file with a rudimentary change log at the top of the file, people have added small changes throughout the years, often without bumping the version number.
 
-This repo is an attempt at version control, using [@AntonioND](http://github.com/AntonioND) fork as baseline.
+This repo has become the new official reference for `hardware.inc`, using [@AntonioND](http://github.com/AntonioND)'s fork as the baseline.
+
+## Contributing
+
+This repository's `master` branch should be considered production.
```
Comment (avivace) [inline README.md:13]:
```suggestion
This repository's `master` branch should be considered production:
```

File: README.md [hunk 8]
```diff
@@ -1,18 +1,28 @@
...
+That is, each commit represents a new release, which thus needs the embedded version number bumped (and an entry added to the changelog at the beginning of the file).
```
Comment (avivace) [inline README.md:14]:
```suggestion
Each commit represents a new release, requiring the embedded version number to be bumped (and an entry to be added to the changelog at the beginning of the file).
```

File: README.md [hunk 9]
```diff
@@ -1,18 +1,28 @@
...
+
+We follow [semantic versioning](https://semver.org).
```
Comment (avivace) [inline README.md:16]:
```suggestion
We follow [semantic versioning](https://semver.org),
```

File: README.md [hunk 10]
```diff
@@ -1,18 +1,28 @@
...
+That is, breaking changes (such as those in [72ec03f](https://github.com/gbdev/hardware.inc/commit/72ec03f835e72be83a1ef189a4a9eac0fbdf39e2)) increase the major version, backwards-compatible changes (typically additions) only increase the minor version, and bugfixes only increase the patch level.
```
Comment (avivace) [inline README.md:17]:
```suggestion
Breaking changes (such as those in [72ec03f](https://github.com/gbdev/hardware.inc/commit/72ec03f835e72be83a1ef189a4a9eac0fbdf39e2)) increase the major version, backwards-compatible changes (typically additions) only increase the minor version, and bugfixes only increase the patch level.
```

**Review by avivace — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (coltongit) [reply README.md:16]: Wouldn't the period be more grammatically correct in this case?

Comment (ISSOtm) [reply README.md:16]: A semicolon, actually.

Comment (coltongit) [reply README.md]: Updated to be the semicolon instead as per your suggestion, ISSO.

**Pull request merged**